airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/default_stream.py

```diff
@@ -44,6 +46,10 @@ def generate_partitions(self) -> Iterable[Partition]:
     def name(self) -> str:
         return self._name
 
+    @property
+    def namespace(self) -> Optional[str]:
```

Comment:
Where is this used? Should it be part of the parent interface?

Reply:
Now that we more commonly instantiate this cursor outside the default stream, this was also reflected in tests. It's used in thread_based_concurrent_stream_scenarios.py for setup [here](https://github.com/airbytehq/airbyte/pull/35905/files#diff-5b143c19641fe03e9fb6369261b27e770e4c0eac8bc6aca1fbc2d2bf22a9dc07R52)

Reply:
So is this only for testing? My understanding is that `thread_based_concurrent_stream_scenarios` is only used for tests. It seems odd to add something that isn't used in the production code base for that

Reply:
Maybe a little odd? But the `namespace` is a valid field on the `DefaultStream` and we need to specify a namespace when instatiating a final state cursor so we have 3 options:
1. Make `namepace` field public
2. Access via the `_namespace` private field
3. In the test scenario builder, assign namespace to  `None`

I'm at least partial to the first two because it stops a small gotcha if we do create stream w/ namespace (even if its not a real use case) and we forget to change the cursor. IMO, even though we added it for a test, its a real getter for a real field so its not crazy that this could be used in the future